File: src/main/java/xyz/devvydont/smprpg/items/blueprints/block/runes/RuneDivinity.java
package xyz.devvydont.smprpg.items.blueprints.block.runes;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import org.bukkit.inventory.ItemStack;
import xyz.devvydont.smprpg.block.CustomBlock;
import xyz.devvydont.smprpg.items.CustomItemType;
import xyz.devvydont.smprpg.items.blueprints.block.BlockBlueprint;
import xyz.devvydont.smprpg.items.interfaces.IFooterDescribable;
import xyz.devvydont.smprpg.services.ItemService;
import xyz.devvydont.smprpg.util.formatting.ComponentUtils;

import java.util.List;

public class RuneDivinity extends BlockBlueprint implements IFooterDescribable {

    public RuneDivinity(ItemService itemService, CustomItemType type) {
        super(itemService, type);
    }

    @Override
    public CustomBlock getCustomBlock() {
        return CustomBlock.RUNE_DIVINITY;
    }

    @Override
    public List<Component> getFooter(ItemStack itemStack) {
        return List.of(
                ComponentUtils.create("When 4 Runes are placed under an Enchanting Table:", NamedTextColor.LIGHT_PURPLE),
                ComponentUtils.merge(
                        ComponentUtils.create("Unlocks the ability to apply ", NamedTextColor.GRAY),
                        ComponentUtils.create("Blessings ", NamedTextColor.YELLOW),
                        ComponentUtils.create("to items.", NamedTextColor.GRAY)
                )
        );
    }
}

File: src/main/java/xyz/devvydont/smprpg/items/blueprints/block/runes/RuneDivinity.kt
package xyz.devvydont.smprpg.items.blueprints.block.runes

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.inventory.ItemStack
import xyz.devvydont.smprpg.block.CustomBlock
import xyz.devvydont.smprpg.items.CustomItemType
import xyz.devvydont.smprpg.items.blueprints.block.BlockBlueprint
import xyz.devvydont.smprpg.items.interfaces.IFooterDescribable
import xyz.devvydont.smprpg.services.ItemService
import xyz.devvydont.smprpg.util.formatting.ComponentUtils

class RuneDivinity(itemService: ItemService, type: CustomItemType) : BlockBlueprint(itemService, type),
    IFooterDescribable {
    override fun getCustomBlock(): CustomBlock {
        return CustomBlock.RUNE_DIVINITY
    }

    override fun getFooter(itemStack: ItemStack?): MutableList<Component?> {
        return mutableListOf(
            ComponentUtils.create(
                "When 4 Runes are placed under an Enchanting Table (3x3):",
                NamedTextColor.LIGHT_PURPLE
            ),
            ComponentUtils.merge(
                ComponentUtils.create("Unlocks the ability to apply ", NamedTextColor.GRAY),
                ComponentUtils.create("Blessings ", NamedTextColor.YELLOW),
                ComponentUtils.create("to items.", NamedTextColor.GRAY)
            )
        )
    }
}
